source/Core/Drivers/FUSB302/policy_engine.cpp

```diff
@@ -617,6 +619,18 @@ void PolicyEngine::PPSTimerCallback() {
   }
 }
 
+bool PolicyEngine::NegotiationTimeoutReached() {
+    if (systemSettings.PDNegTimeout == 0){
+	return false;
+    }
+
+    if (xTaskGetTickCount() > (TICKS_SECOND/10 * systemSettings.PDNegTimeout)) {
```

Comment:
You will also need to take into consideration the 1-2 second bootloader time of the TS80P most likely if you need hard time since boot

Reply:
I think it should be addressed by the setting itself, so if it was set to 1 and bootloader took 1.2 seconds to finish, then it will wait for 1.3 s just like if it was set to 13 without additional wait for bootloader.

If the extra 100 ms wait matters, it may be initialized to -1 for disabled state (and modify variable to signed int) or a higher (special) value, so that way it could be solved for the best possible way of practically instant action (so when main software starts).

Reply:
I don't know if it matters at all, was just noting it might be something to take into awareness, since the "0" mark of xTaskGetTickCount() is once the bootloader finishes, hardware init finishes and we kick off the RToS.

Though I dont _think_ it will matter.

Reply:
Do you mean it may check against a more native tick counter or timer other than the task's counter?
I don't know if it has a built-in independent counter anyway or a free timer which could be started.

...or you mean GUIThread is started after POWThread and if a call happens from the latter to the former, it may run to an uninitialized variable as task's tick counter?

POWThread calls vTaskDelay(TICK_100MS) prior running power_check(), and GUIThread does not have such a delay and it should finish initialization in 100 ms I think.
Also, timeout's minimal value is currently 1, which also adds an additional 100 ms delay, so I don't think it should be a problem if that delay remains there...

I think OLED's initialization over I2C may be the slowest part of initialization, but as far as I can see, display comes up quickly at least on Pinecil.